main.go

```diff
@@ -68,7 +88,7 @@ func run(ctx context.Context) error {
 		}
 
 		cancel()
-
+		otelShutdown(ctx)
```

Comment:
I would expect to see all shutdown logic before the shutdown timeout is cancelled. I would suggest the best way to address this would be to move the `cancel()` call to line 83 as `defer cancel()`. This will ensure the code is more maintainable by preventing others adding their code after the `cancel()` as you have. This is a pattern we favour within our other apps.